test/tools/dshell_prebuilt/dshell_prebuilt.d

```diff
@@ -0,0 +1,296 @@
+*/
+auto tryRun(scope const(char[])[] args, File stdout = std.stdio.stdout, string[string] env = null)
+{
+    std.stdio.stdout.write("[RUN]");
+    if (env)
+    {
+       foreach (pair; env.byKeyValue)
+       {
+           std.stdio.stdout.write(" ", pair.key, "=", pair.value);
+       }
+    }
+    std.stdio.write(" ", escapeShellCommand(args));
+    if (stdout != std.stdio.stdout)
+    {
+        std.stdio.stdout.write(" > ", stdout.name);
+    }
+    std.stdio.stdout.writeln();
+    std.stdio.stdout.flush();
+    auto proc = spawnProcess(args, stdin, stdout, std.stdio.stderr, env);
+    return wait(proc);
+}
+/// ditto
+void run(scope const(char[])[] args, File stdout = std.stdio.stdout, string[string] env = null)
+{
+    const exitCode = tryRun(args, stdout, env);
+    if (exitCode != 0)
+    {
+        writefln("Error: last command exited with code %s", exitCode);
+        assert(0, "last command failed");
+    }
+}
+/// ditto
+void run(string cmd, File stdout = std.stdio.stdout, string[string] env = null)
+{
+    // TODO: option to disable this?
+    if (SEP != "/")
+        cmd = cmd.replace("/", SEP);
+    run(parseCommand(cmd), stdout, env);
+}
+
+/**
+Parse the given string `s` as a command.  Performs BASH-like variable expansion.
+*/
+string[] parseCommand(string s)
+{
+    auto rawArgs = s.split();
+    auto args = appender!(string[])();
+    foreach (rawArg; rawArgs)
+    {
+        args.put(shellExpand(rawArg));
+    }
+    return args.data;
+}
+
+/// Expand the given string using BASH-like variable expansion.
+string shellExpand(const(char)[] s)
+{
+    auto expanded = appender!(char[])();
+    for (size_t i = 0; i < s.length;)
+    {
+        if (s[i] != '$')
+        {
+            expanded.put(s[i]);
+            i++;
+        }
+        else
+        {
+            i++;
+            assert(i < s.length, "lone '$' at end of string");
+            auto start = i;
+            if (s[i] == '{')
+            {
+                start++;
+                for (;;)
+                {
+                    i++;
+                    assert(i < s.length, "unterminated ${...");
+                    if (s[i] == '}') break;
+                }
+                expanded.put(Vars.get(s[start .. i]));
+                i++;
+            }
+            else
+            {
+                assert(validVarChar(s[i]), "invalid sequence $'" ~ s[i]);
+                for (;;)
+                {
+                    i++;
+                    if (i >= s.length || !validVarChar(s[i]))
+                        break;
+                }
+                expanded.put(Vars.get(s[start .. i]));
+            }
+        }
+    }
+    auto result = expanded.data;
+    return (result is null) ? "" : result.assumeUnique;
+}
+
+// [a-zA-Z0-9_]
+private bool validVarChar(const char c)
+{
+    if (c < 'A')
+        return (c >= '0' && c <= '9');
+    if (c <= 'Z')
+        return true;
+    if (c == '_')
+        return true;
+    return (c >= 'a' && c <= 'z');
```

Comment:
Why not 

```d
import std.ascii : isAlphaNum;
return c.isAlphaNum || c == '_';
```

https://dlang.org/phobos/std_ascii.html#isAlphaNum

Reply:
Good suggestion. Fixed.